### PR TITLE
FAQ

### LINKED ISSUE #16: FAQ Section
Database schema will be the same as the existing FAQ section, but will be a different collection.

Search should update as you type. Find good library :)

image

### PR DESCRIPTION
**Pull request by jackyzha0**
Closes #16 

# stuff
* fetches from firestore root FAQ collection (selects current hackathon and livesite specific ones)
* two column responsiveness
* debounced search for faq

# Filter in action
image

# Full thing
image

# Mobile (thin) view
image

### CODE REVIEW

**Review by ianmah — COMMENTED** [review]
Can we make the FAQ card have a pointer mouse

File: src/components/Faq.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  & > li {
+    ${CardLike};
+    padding: 0.5em 1em;
+    margin-bottom: 1em;
+    list-style-type: none;
+  }
+`
+
+export default ({ faq }) => {
+  const categories = faq.reduce((q, a) => {
```
Comment (ianmah) [inline src/components/Faq.js:36]: is q question and a answer? Can we use longform as variable names instead?

File: src/components/Faq.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  function createFAQList(entries) {
+    return (
+      <DetailFlex>
+        {
+          splitHalf(entries).map(half =>
+            <DetailColumn>{half.map(singleEntry)}</DetailColumn>
+          )
+        }
+      </DetailFlex>
```
Comment (ianmah) [inline src/components/Faq.js:54]: is this to make two columns of FAQ boxes?

File: src/components/Faq.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    );
+  }
+
+  function singleEntry(entry) {
+    return (
+      <li>
+        <Accordion heading={entry.question}>
+          <P>{entry.answer}</P>
```
Comment (ianmah) [inline src/components/Faq.js:62]: I think this could use more padding on the bottom

File: src/components/Accordion.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+`
+
+export default ({ heading, children }) => {
+  return (
+    <Accordion>
+      <summary>
+        <H3>{heading}</H3>
+      </summary>
+      {children}
+    </Accordion>
```
Comment (ianmah) [inline src/components/Accordion.js:24]: the design has the arrow on the right, do you think that violates any best-practices/a11y

File: src/components/Faq.js (hunk 2 above)
Comment (jackyzha0) [reply src/components/Faq.js:54]: yes

File: src/components/Accordion.js (hunk 4 above)
Comment (jackyzha0) [reply src/components/Accordion.js:24]: image

From what I've seen, left-side is generally best practice but I can change it to design if necessary

File: src/components/Faq.js (hunk 2 above)
Comment (ianmah) [reply src/components/Faq.js:58]: Any reason why we can't use flexbox wrap instead?

Comment (jackyzha0) [reply src/components/Faq.js:58]: Then questions might get inadvertently shuffled around once you expand enough Qs, I tried this and the UX is kinda nasty

**Review by ianmah — APPROVED** [review]

**Pull request merged**